src/demetriek/exceptions.py

```diff
@@ -15,3 +20,28 @@ class LaMetricAuthenticationError(LaMetricError):
 
 class LaMetricConnectionTimeoutError(LaMetricConnectionError):
     """LaMetric connection Timeout exception."""
+
+
+def raise_on_data_error(
```

Comment:
We don't need to extract this, we can just do that inline (as it is only used once).